channels/applicationinsights-channel-js/src/Statsbeat.ts

```diff
@@ -0,0 +1,203 @@
+                return !!_isEnabled;
+            }
+
+            _self.countRequest = (endpoint: string, duration: number, success: boolean) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.totalRequestCount++;
+                counter.intervalRequestExecutionTime += duration;
+                if (success === false) {
+                    counter.totalFailedRequestCount++;
+                }
+                else {
+                    counter.totalSuccesfulRequestCount++;
+                }
+            }
+
+            _self.countException = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.exceptionCount++;
+            }
+            
+            _self.countThrottle = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.throttleCount++;
+            }
+
+            _self.countRetry = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.retryCount++;
+            }
+
+            _self.trackShortIntervalStatsbeats = (): void => {
+                _trackRequestDuration();
+                _trackRequestsCount();
+                _sendStatsbeats();
+            }
+
+            function _getCustomProperties() {
+                _cikey = _config.instrumentationKey;
+            }
+
+            function _getNetworkStatsbeatCounter(host: string): NetworkStatsbeat {
+                // Check if counter is available
+                if (_networkCounter && _networkCounter.host === host) {
+                    return _networkCounter;
+                }
+                // Create a new one if not found
+                let newCounter = new NetworkStatsbeat(host);
+                return newCounter;
+            }
+
+            function _sendStatsbeats() {
+                // Add extra properties
+                let networkProperties = {
+                    "cikey": _cikey,
+                }
+                if (objKeys(_statsbeatMetrics)) {
+                    let statsbeat: ITelemetryItem = {
+                        iKey: Statsbeat.INSTRUMENTATION_KEY,
+                        name: Statsbeat.NETWORK,
+                        baseData: {
+                            name: Statsbeat.NETWORK,
+                            average: 0,
+                            properties: {"host": _networkCounter.host, ..._statsbeatMetrics.properties, ...networkProperties},
```

Comment:
Is this logic correct? I believe each entry in `_statsbeatMetrics` is supposed to be sent as a separate metric, not as a custom dimension?

Reply:
Yeah for web sdk, we decide to group the metrics into one telemetry - this would cause the query change. I listed a point in description for this change.